guides/plugins/plugins/storefront/add-translations.md

```diff
@@ -80,3 +80,27 @@ Translation with placeholders:
 $this->trans('soldProducts', ['%count%' => 3, '%country%' => 'Germany']);
 ```
 
+## Using translation generally in PHP
+
+If we need to use a snippet elsewhere in PHP, we can use Dependency Injection to inject `Shopware\Core\Framework\Adapter\Translation\Translator` which implements Symfony's `Symfony\Contracts\Translation\TranslatorInterface`:
+
+```xml
+<service id="Swag\Example\Service\SwagService" public="true" >
+    <argument type="service" id="Shopware\Core\Framework\Adapter\Translation\Translator" />
```

Comment:
I think you should refer to the `translator` service id instead as the adapter here is just a decorator for that service.

Reply:
Done. I hadn't even noticed that myself, I've even seen several store plugins that use the Adapter directly.

Reply:
👀 I don't think that this is a good idea but hey good that the docs are better

Reply:
Me neither. But when the core itself uses the Adapter (https://github.com/shopware/core/blob/trunk/Checkout/DependencyInjection/document.xml#L71), that's what you find in the code and end up copying when the docs don't mention it 😄

Reply:
Will you or do I PR that away?

Reply:
I found it quite a few places, I'll make a PR that removes it 😄

Reply:
Link it to me :D

Reply:
PR is here:
https://github.com/shopware/platform/pull/1884